libcob/common.h

```diff
@@ -1294,6 +1294,18 @@ struct cob_call_struct {
 	cob_call_union		cob_cstr_cancel;	/* Cancel entry */
 };
 
+#ifdef HAVE_JNI
+typedef struct __cob_java_static_method {
+	jclass cls;
+	jmethodID mid;
+} cob_java_handle;
```

Comment:
Here the definition requires something like
```c
#ifdef HAVE_JNI
#include "jni.h"
#endif 
```
Otherwise, that definition could actually become abstract (as below) if we only ever manipulate pointer to its value outside of `java.c`, and in that case the inclusion of `jni.h` is not needed (which may be desirable).
```c
typedef struct __cob_java_handle cob_java_handle;
```

Reply:
I was going to suggest for now we should include jni.h inside the header guard, as we fix the config and check for other issues. However having an abstract definition definitely is a better option, I did check it `cob_java_handle` is only used in java.c